### PR TITLE
Add extra_flags_per_feature to cc toolchain

### PR DESCRIPTION
**Pull request by hofbi**
Closes https://github.com/tweag/rules_nixpkgs/issues/574

### CODE REVIEW

File: toolchains/cc/cc.nix [hunk 1]
```diff
@@ -298,6 +298,7 @@ pkgs.runCommand "bazel-${cc.orignalName or cc.name}-toolchain"
         if [[ -x ${cc}/bin/ld.gold ]]; then echo True; else echo False; fi
       )
     )
+    COVERAGE_LINK_FLAGS=()
```
Comment (aherrmann) [inline toolchains/cc/cc.nix:301]: why is this needed? It overrides the [set of flags defined above](https://github.com/tweag/rules_nixpkgs/pull/575/files#diff-ebda446a0ba42acd64ea7fe331d17801b2c3f88c96e3fe1d785f95345404523dR288).

Comment (hofbi) [reply toolchains/cc/cc.nix:301]:
```suggestion
    EXTRA_FLAGS_PER_FEATURE=()
```

**Review by aherrmann — APPROVED** [review]
Thank you!

File: toolchains/cc/cc.nix [hunk 2]
```diff
@@ -298,6 +298,7 @@ pkgs.runCommand "bazel-${cc.orignalName or cc.name}-toolchain"
         if [[ -x ${cc}/bin/ld.gold ]]; then echo True; else echo False; fi
       )
     )
+    EXTRA_FLAGS_PER_FEATURE=()
```
Comment (Silic0nS0ldier) [inline toolchains/cc/cc.nix:301]: Discovered while testing that Bazel expects a `dict` here. Noted at https://github.com/tweag/rules_nixpkgs/issues/574#issuecomment-2314438748.

`tool_paths` is an example of a `dict` whose value is set here, the problem is that is a bash associative array (it is represented as a dict as that is the only way it can be).

It may be necessary to handle the case where `repr` returns `[]` in `toolchains/cc/cc.bzl` to fix this.

Comment (aherrmann) [reply toolchains/cc/cc.nix:301]: See also https://github.com/tweag/rules_nixpkgs/pull/576
